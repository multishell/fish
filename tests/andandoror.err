
####################
# Basic && and || support

####################
# && and || in if statements

####################
# && and || in while statements

####################
# Nots

####################
# Complex scenarios
