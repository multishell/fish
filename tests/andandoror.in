logmsg "Basic && and || support"

echo first && echo second
echo third || echo fourth
true && false ; echo "true && false: $status"
true || false ; echo "true || false: $status"
true && false || true ; echo "true && false || true: $status"

logmsg "&& and || in if statements"

if true || false ; echo "if test 1 ok" ; end
if true && false ; else; echo "if test 2 ok" ; end
if true && false ; or true ; echo "if test 3 ok" ; end
if [ 0 = 1 ] || [ 5 -ge 3 ] ; echo "if test 4 ok"; end

logmsg "&& and || in while statements"

set alpha 0
set beta 0
set gamma 0
set delta 0
while [ $alpha -lt 2 ] && [ $beta -lt 3 ]
      or [ $gamma -lt 4 ] || [ $delta -lt 5 ]
    echo $alpha $beta $gamma
    set alpha ( math $alpha + 1 )
    set beta ( math $beta + 1 )
    set gamma ( math $gamma + 1 )
    set delta ( math $delta + 1 )
end

logmsg "Nots"

true && ! false ; echo $status
not true && ! false ; echo $status
not not not true ; echo $status
not ! ! not true ; echo $status
not ! echo not ! ; echo $status

logmsg "Complex scenarios"

begin; echo 1 ; false ; end || begin ; echo 2 && echo 3 ; end

if false && true
   or not false
   echo 4
end
