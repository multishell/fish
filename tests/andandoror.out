
####################
# Basic && and || support
first
second
third
true && false: 1
true || false: 0
true && false || true: 0

####################
# && and || in if statements
if test 1 ok
if test 2 ok
if test 3 ok
if test 4 ok

####################
# && and || in while statements
0 0 0
1 1 1
2 2 2
3 3 3
4 4 4

####################
# Nots
0
1
1
0
not !
0

####################
# Complex scenarios
1
2
3
4
